NASA-AMMOS/aerie-ui
# Pull from constraint run table

Resolves #770 

This PR introduces a subscription to the `constraint_run` table to pull the last cached constraint check status for the latest simulation dataset.

Ideally the behavior should be exactly the same for displaying constraint statuses, but the only difference would be that the last constraint status is persisted between reloads.

To test:
1. Create a constraint `export default (): Constraint => Real.Resource("/fruit").lessThan(4)`
2. Create banananation plan
3. Add a `GrowBanana`
4. Associate the constraint created in step 1 to the plan
5. Simulate
6. Check constraints
7. Verify that the constraint status is all green
8. Refresh and verify that it still shows as all green
9. Add another `GrowBanana`
10. Simulate again
11. Check constraints again
12. Verify that a validation error shows
13. Refresh and verify that it still shows the validation error
